dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/generate/EndpointsEnumGenerator.java

```diff
@@ -0,0 +1,142 @@
+                        TypeName.get(Optional.class),
+                        serviceDefinition.serviceInterface())
+                .build());
+
+        return enumBuilder.build();
+    }
+
+    private TypeSpec endpointField(EndpointDefinition def, String serviceName, Optional<String> apiVersion) {
+        TypeSpec.Builder builder = TypeSpec.anonymousClassBuilder("");
+
+        builder.addField(FieldSpec.builder(
+                                TypeName.get(PathTemplate.class), "pathTemplate", Modifier.PRIVATE, Modifier.FINAL)
+                        .initializer(pathTemplateInitializer(def.httpPath()))
+                        .build())
+                .addMethod(MethodSpec.methodBuilder("renderPath")
+                        .addAnnotation(Override.class)
+                        .addModifiers(Modifier.PUBLIC)
+                        .addParameter(ParameterizedTypeName.get(Map.class, String.class, String.class), "params")
+                        .addParameter(UrlBuilder.class, "url")
+                        .addCode("pathTemplate.fill(params, url);")
+                        .build())
+                .addMethod(MethodSpec.methodBuilder("httpMethod")
+                        .addAnnotation(Override.class)
+                        .addModifiers(Modifier.PUBLIC)
+                        .returns(HttpMethod.class)
+                        .addCode(CodeBlock.builder()
+                                .add("return $T.$L;", HttpMethod.class, def.httpMethod())
+                                .build())
+                        .build())
+                .addMethod(MethodSpec.methodBuilder("serviceName")
+                        .addAnnotation(Override.class)
+                        .addModifiers(Modifier.PUBLIC)
+                        .returns(String.class)
+                        .addCode(CodeBlock.builder()
+                                .add("return $S;", serviceName)
+                                .build())
+                        .build())
+                .addMethod(MethodSpec.methodBuilder("endpointName")
+                        .addAnnotation(Override.class)
+                        .addModifiers(Modifier.PUBLIC)
+                        .returns(String.class)
+                        .addCode(CodeBlock.builder()
+                                .add("return $S;", def.endpointName().get())
+                                .build())
+                        .build())
+                .addMethod(MethodSpec.methodBuilder("version")
+                        .addAnnotation(Override.class)
+                        .addModifiers(Modifier.PUBLIC)
+                        .returns(String.class)
+                        .addCode(CodeBlock.builder()
+                                .add(apiVersion
+                                        .map(s -> CodeBlock.of("return $S;", s))
+                                        .orElseGet(() -> CodeBlock.of("return VERSION;")))
+                                .build())
```

Comment:
I think this becomes
```suggestion
                        .addCode(CodeBlock.of("return VERSION;"))
```